src/cheri_insts.sail

```diff
@@ -916,7 +916,7 @@ function clause execute (CBuildCap(cd, cs1, cs2)) = {
       RETIRE_FAIL
     } else {
       assert(exact, "CBuildCap: setCapBounds was not exact"); /* base and top came from cs2 originally so will be exact */
-      let cd5 = if signed(cs2_val.otype) == otype_sentry then sealCap(cd4, to_bits(cap_otype_width, otype_sentry)) else cd4;
+      let cd5 = if signed(cs2_val.otype) == otype_sentry & cs2_val.permit_execute then sealCap(cd4, to_bits(cap_otype_width, otype_sentry)) else cd4;
```

Comment:
Should this raise a permit execute violation?

Reply:
Also could do with some line breaks.

Reply:
Possibly, I'm just matching https://github.com/CTSRD-CHERI/qemu/commit/cd35dc7102d54fd36b7761997318bcbfe9cd43cb here. Not sure what hardware does in this case.

Reply:
Toooba does not check for the executable permission on sentry caps. There is an `isDerivable` check but it only checks the exponent is in range, the reserved bits are zero and the top and base bits look sensible. Maybe it should be the job of `CCopyType` and `CCSeal` to restore special sealed capabilities like sentries and, in future, indirect sentries?

In an ideal world we would use the Sail to prove that `CBuildCap`, `CCopyType` and `CCSeal` can only be used to create capabilities that are derivable by other instructions and that any derivable instruction can be restored from bits :-)

Reply:
Doing it with CCSeal does not make sense. CSeal cannot make sentries, and CCSeal is just a conditional version of it. Nor should they, there is no authorising cap. Having it be CBuildCap was a very deliberate choice.

It also means it's only one instruction for every function pointer you need to initialise rather than three.

Reply:
And yeah, we should be throwing an exception here, leaving it unsealed but valid is a bad idea

Reply:
Just wondering what goes wrong if we give up enforcing this: hopefully all the sail code checks the permit_execute bit when jumping to sentries anyway.

Reply:
See https://github.com/CTSRD-CHERI/cheri-architecture/issues/40 for the discussion we had last time about this.  I broadly agree that, until we find architectural purpose and a non-cbuildcap way of constructing such capabilities, that cbuildcap shouldn't construct them, but I think it would also be fine to clear the tag on the cbuildcap result for otype==sentry and !Permit_Execute rather than raising an exception.

Reply:
In a tag-clearing world, yeah, you'd tag-clear here, but that's not what QEMU does, it just skips the sealing step...

Reply:
Certainly skipping the sealing step is a bug, yes.